crates/core/src/limits.rs

```diff
@@ -27,6 +27,15 @@ impl ResourceLimiterAsync for StoreLimitsAsync {
         if can_grow {
             self.memory_consumed =
                 (self.memory_consumed as i64 + (desired as i64 - current as i64)) as u64;
+        } else {
+            tracing::warn!(
+                error.type_ = "memory_limit_exceeded",
+                current,
+                desired,
+                _maximum,
+                max_memory_size = self.max_memory_size,
+                "Memory growth denied",
```

Comment:
```suggestion
                "instance memory limit exceeded",
```
Might be a little more clear.